File: ios-native/ClubFuoco/Models/UserProfile.swift
import Foundation
import SwiftUI

/// Account type drives which tab set the app shows (see BottomNav.tsx).
/// Unknown values fall back to .user, matching AuthContext's behavior.
enum AccountType: String, Decodable, Sendable {
    case user, club, dj

    init(from decoder: Decoder) throws {
        let raw = try decoder.singleValueContainer().decode(String.self)
        self = AccountType(rawValue: raw) ?? .user
    }
}

/// Self-declared gender, required at signup. Drives guest-list payout
/// settlement (clubs pay promoters different per-head rates by gender);
/// `preferNotToSay` settles at the higher rate as the legal default.
enum Gender: String, CaseIterable, Sendable, Identifiable {
    case male
    case female
    case preferNotToSay = "prefer_not_to_say"

    var id: String { rawValue }
}

/// Row from public.users (`select('*')` — we model only the fields the app
/// reads; Codable ignores the rest). Dates stay String: Postgres date columns
/// arrive as "YYYY-MM-DD", timestamps as ISO 8601 with variable precision.
struct UserProfile: Decodable, Sendable {
    let id: UUID
    let email: String?
    let fullName: String?
    let phone: String?
    let birthday: String?
    let gender: String?
    let accountType: AccountType?
    let role: String?
    let stripeCustomerId: String?
    let createdAt: String?

    /// Same completeness rule routeAfterOAuth() applies before allowing the
    /// user past /complete-profile. Gender was added 2026-06-22 for guest-list
    /// payout settlement — existing rows without one re-enter complete-profile
    /// on next launch (see RootView's gate).
    var isComplete: Bool {
        fullName?.isEmpty == false && email?.isEmpty == false
            && phone?.isEmpty == false && birthday?.isEmpty == false
            && gender?.isEmpty == false
    }
}

/// Three-pill selector used by the signup wizard and complete-profile screens.
/// Stays here to keep gender UI co-located with the enum.
struct GenderPicker: View {
    @Binding var selection: Gender?
    let locale: LocaleStore

    var body: some View {
        // Two-row layout: Male/Female share the top row at equal width;
        // "Prefer not to say" gets its own full-width row underneath so its
        // longer label never clips. Order matches Gender.allCases.
        VStack(spacing: 8) {
            HStack(spacing: 8) {
                chip(.male)
                chip(.female)
            }
            chip(.preferNotToSay)
        }
    }

    @ViewBuilder
    private func chip(_ g: Gender) -> some View {
        let isSelected = selection == g
        Button {
            selection = g
            Haptics.tap()
        } label: {
            Text(locale.t("signup.gender.\(g.rawValue)"))
                .font(.cfSans(13, weight: .medium))
                .foregroundStyle(isSelected ? Theme.cream : Theme.ink)
                .frame(maxWidth: .infinity)
                .frame(height: 44)
                .background(isSelected ? Theme.wine : Color.white, in: .rect(cornerRadius: 11))
                .overlay(RoundedRectangle(cornerRadius: 11)
                    .stroke(isSelected ? Theme.wine : Theme.hairline))
        }
        .buttonStyle(.plain)
    }
}

File: android-native/app/src/main/java/com/clubfuoco/app/models/UserProfile.kt
package com.clubfuoco.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Account type drives which tab set the app shows. Unknown values fall back to
 * `user`, matching AuthContext on web and `AccountType` on iOS.
 */
enum class AccountType {
    USER, CLUB, DJ;

    companion object {
        fun from(raw: String?): AccountType =
            entries.firstOrNull { it.name.equals(raw, ignoreCase = true) } ?: USER
    }
}

/**
 * Self-declared gender, required at signup. Drives guest-list payout settlement
 * (clubs pay promoters different per-head rates by gender); `prefer_not_to_say`
 * settles at the higher rate as the legal default.
 */
enum class Gender(val raw: String) {
    MALE("male"),
    FEMALE("female"),
    PREFER_NOT_TO_SAY("prefer_not_to_say");

    /** Key into the string catalog: `signup.gender.<raw>` → `signup_gender_<raw>`. */
    val stringKey: String get() = "signup_gender_$raw"

    companion object {
        fun from(raw: String?): Gender? = entries.firstOrNull { it.raw == raw }
    }
}

/**
 * Row from `public.users`. Only the fields the app reads are modelled; the
 * serializer ignores the rest.
 *
 * Dates stay String: Postgres date columns arrive as "YYYY-MM-DD" and
 * timestamps as ISO 8601 with variable precision, so parsing them eagerly buys
 * nothing and loses fidelity.
 */
@Serializable
data class UserProfile(
    val id: String,
    val email: String? = null,
    @SerialName("full_name") val fullName: String? = null,
    val phone: String? = null,
    val birthday: String? = null,
    val gender: String? = null,
    @SerialName("avatar_url") val avatarUrl: String? = null,
    @SerialName("account_type") val accountType: String? = null,
    /** 'user' | 'promoter' — promoters cannot use the consumer app. */
    @SerialName("account_kind") val accountKind: String? = null,
    val role: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
) {
    val resolvedAccountType: AccountType get() = AccountType.from(accountType)

    /**
     * The same completeness rule `routeAfterOAuth()` applies before letting the
     * user past complete-profile.
     *
     * Gender was added 2026-06-22 for guest-list payout settlement, so existing
     * rows without one re-enter complete-profile on next launch. Phone is
     * deliberately OPTIONAL (2026-07-05) — gating on it bounced phone-less users
     * into complete-profile on every single launch.
     */
    val isComplete: Boolean
        get() = !fullName.isNullOrEmpty() &&
            !email.isNullOrEmpty() &&
            !birthday.isNullOrEmpty() &&
            !gender.isNullOrEmpty()
}
